CONTRIBUTING.md

```diff
@@ -16,128 +16,158 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
 
 > **Important:**
 > PRs from forks of this repository are likely to fail automated workflows due to access to secrets. PRs from forks will be considered but may experience additional delay for testing.
 
-### Rules
+### General rules
 
-* minimal prose
-* minimalist code
-* no azureml-* in training code
-* examples (including notebooks) can be re-run without failing in less than 10 minutes
-* tutorials must be re-run without failing at least daily
-* `pip install --upgrade -r requirements.txt` remains <60s
+- minimal prose
+- minimalist code
+- examples and notebooks can be re-run without failing in less than 10 minutes
+- tutorials can re-run without failing in less than 3 hours
 
-### Checks
+### Miscellaneous
+
+- to modify `README.md`, you need to modify `readme.py` and accompanying markdown files
+- the tables in the `README.md` are auto-generated, including description, via other files
+- develop on a branch, not a fork, for workflows to run properly
+- use an existing environment where possible
+- use an existing dataset where possible
+- don't register environments
+- don't create compute targets
+- don't modify `requirements.txt`
+- you probably shouldn't modify any files in the root of the repo
+- you can `!pip install --upgrade packages` as needed in notebooks
+- you can (and likely should) abstract setup for tutorials in a `setup.sh` file or similar
+
+### Modifying an existing example
 
 If modifying existing examples, before a PR:
 
-* run `python readme.py` from the root of the repo
-* this will generate the `README.md` file
-* this will generate the `run-examples` and `run-notebooks` workflow files
-* this will format Python code and notebooks
+- run `python readme.py` from the root of the repo
+- this will generate the `README.md` file
+- this will generate the `run-examples` and `run-notebooks` workflow files
+- this will format Python code and notebooks
 
-If you are adding new examples, see below.
+### Enforced naming
 
-### Organization
+Directories and files must follow:
 
-PRs to add new examples should consider which type of example to add:
+- naming must be logical
+- directories under `tutorials` must be words separated by hyphens
+- tutorial workflows (and workflow files) use the naming convention `run-tutorial-*initials*`, where *initials* is the initials of the words
 
-* `examples` is for general examples using AML and should run `code` examples
-* `notebooks` is for general example notebooks using AML and should be interactive
-* `tutorials` is for end to end tutorials using AML
+### Unenforced naming
 
-### Enforced naming
+- `environment_name` = "framework-example|tutorial" e.g. "pytorch-example"
+- `experiment_name` = "logical-words-example|tutorial" e.g. "hello-world-tutorial"
+- `compute_name` = "compute-defined-in-setup-workspace.py" e.g. "gpu-K80-2"
+- `ws = Workspace.from_config()`
+- `dstore = ws.get_default_datastore()`
+- `ds = Dataset.File.from_files(...)`
```

Comment:
What if there are multiple datasets?

Reply:
`dss` for a list of datasets, `ds1`, `ds2`, `ds3`, etc. for individual datasets, would be my opinion but this is in the unenforced naming section

same for the other objects